pubsub/google/cloud/pubsub_v1/subscriber/_consumer.py

```diff
@@ -341,10 +307,12 @@ def _blocking_consume(self, policy):
                 _LOGGER.debug('Event signaled consumer exit.')
                 break
 
-            request_generator = self._request_generator_thread(policy)
-            response_generator = policy.call_rpc(request_generator)
-            responses = _pausable_iterator(
-                response_generator, self._can_consume)
+            initial_request = policy.get_initial_request()
+            request_generator = _RequestQueueGenerator(
+                self._request_queue, initial_request=initial_request)
+            rpc = policy.call_rpc(request_generator)
+            request_generator.rpc = rpc
```

Comment:
Ugh, that looks racy, at least on the surface.  I guess we can't get around it, though.